packages/toolshed/routes/ai/llm/llm.routes.ts

```diff
@@ -7,31 +7,26 @@ import {
   type LLMGenerateObjectRequest,
   type LLMMessage,
   type LLMRequest,
-  LLMTypedContent,
+  type LLMTypedContentPart,
 } from "@commontools/llm/types";
 
 const tags = ["AI Language Models"];
 
-const TypedMessageSchema = toZod<LLMTypedContent>().with({
-  type: z.enum(["text", "image"]),
-  data: z.string(),
+const MessageContentSchema = z.object({
+  type: z.enum(["text", "image", "tool-call", "tool-result"]),
+  text: z.string().optional(),
+  image: z.string().url().optional(),
+  toolCall: z.string().optional(),
+  toolResponse: z.string().optional(),
 });
```

Comment:
Given that we are re-declaring this, I think we can indeed change our other types back to be a proper union of operations.